server/types/uof.d.ts

```diff
@@ -24,18 +40,53 @@ type GetUsersResults = {
   staffId?: number // only if exists
 }
 
+export interface PrisonLocation {
+  agencyId?: string
+  currentOccupancy?: number
+  description?: string
+  internalLocationCode?: string
+  locationId?: number
+  locationPrefix?: string
+  locationType?: string
+  locationUsage?: string
+  operationalCapacity?: number
+  parentLocationId?: number
+  userDescription?: string
+}
+
 export interface UserService {
   /** get current user */
   getUser: (token: string) => Promise<User>
   /** Get details for users along with email address */
   getUsers: (token: string, usernames: string[]) => Promise<GetUsersResults[]>
 }
 
+export interface OffenderService {
```

Comment:
Think it would probably be good to get the services/clients themselves to export their own types. 
So this could probably live in OffenderService. Not too sure but could even derive it based on the service object, or maybe turn the service object into a class?

(As reporting client is directly returned (and it's not a factory) it's not strictly needed to define an interface for it merely for type safety - the type can just be imported directly)

Reply:
Sounds good to me. 
I couldn't decide if clients should export their types, or if the 'domain' should define that are then implemented by the clients.  If the types and interfaces are part of the domain then the implementations should depend upon those rather then the domain depending upon the implementations...